python-corsikaio@0.6.0: 1 test fails with assertion equal (?)

Hi,

I've Notised during Guix Astro Update 2025/12 preparation:

Log tail:
```
________________________________ test_versions _________________________________

    def test_versions():
        from corsikaio.io import read_buffer_size, read_block
        from corsikaio.subblocks import get_version
        from corsikaio.constants import RUNH_VERSION_POSITION
        from corsikaio.constants import EVTH_VERSION_POSITION

        for path, version in zip(test_files, (6.5, 7.41)):
            buffer_size = read_buffer_size(path)

            with open(path, 'rb') as f:
                block = read_block(f, False, buffer_size)
>               assert get_version(block, RUNH_VERSION_POSITION) == version
E               assert 7.41 == 7.41
E                +  where 7.41 = <function get_version at 0x7ffff13f5bc0>(b'RUNH\x00\x00\x80?\x80\x02\'H\xb8\x1e\xed@\x00\x00\x80?\x00\xd8VH\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xcd\xcc,\xc0\x00\x00\xfaD\x00\x00\xfaD\x00\x00\x80?\x00\x00\x00\x00\x9a\x99\x99>\x9a\x99\x99>\n\xd7\xa3<\n\xd7\xa3<l\xe7\x17N\x00|\x12I\x00$\xf4I\x7f\x10\xa7\xc4\x00\x00\x00\x00\t\x9f;=\x00\xb6\x12?\xb6dX=\x00\x00 @\xe1z\x04@33\x03A\xcd\xcc\xcc=\x00\x00\x00\x00\x00\x00\x00\x00\xc4\x00\x80?\x17z\x1e<\x01\x00\x80?;\xc4\x16:\x00\x00\x00\x00\x00\x00\x00\x00\xcd\xcc\x16B\xbf\xbb 9\xc4.\x16Ao\x12\x03;\xe0\\\xdfP\x00\x00\x80?@Q\n?\xc3\xf5\xc8?}\x1d\x90&1\x08\xac<\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xa0A\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x19\x86>\x89[f?\x00\x00\x00\x00\xb1\xfd\x84?\xc4\xf2\x8a>7\t\tC\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xa0A\x00\x00\x00\x00\x00\x00\x00\x00\xcd\xcc\xcc=\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80>\x00\x00\x00?\x00\x00@?\x00\x00\x80?\x00\x00\x00?\xcd\xccL>\xfc(m?\x92\xe1\x8f?\x9a\x99\x15C\x9a\x99\x15CA/q>\xaa\xf1R>q=\n>\xf8Sc>\x00\x00\x00?\x00\x00\x00\x00|a"?\xdaX1?\xce\xaa_?\x0c\x93)?\x8bl\xc7>}]\x1c@\x17\x93i?\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80?\x00P\xc3G\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\t\xca>\xecQ8?\xfc\x18s?\xc3\xf5\x88\xbf\xe1z\x04@\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00o\xf0%?=,\x04?\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00h\x91-?\xe7\xfbi?\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00@\x9cG\x00\x00\x00\x00\x00\x00\x00\x00\x805=I\x80j\xc9I\x80\x96\x98J\xa07 K,\x91\x10\xc3\xb5)C\xc2x\xbd\xb9>>*\xdf\xb9\xd9\xd8?:&\x02\x95D\xf8\xd7\x91D\xcb\x9e\xafD\x01\x02RD\x00\x00\x80?\xa6\xb8rI<\x07GI\xf1N\x1bI:\xa72I\xcbakP\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00HC', 4)

tests/test_io.py:154: AssertionError
=========================== short test summary info ============================
FAILED tests/test_io.py::test_versions - assert 7.41 == 7.41
 +  where 7.41 = <function get_version at 0x7ffff13f5bc0>(b'RUNH\x00\x00\x80?\x80\x02\'H\xb8\x1e\xed@\x00\x00\x80?\x00\xd8VH\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xcd\xcc,\xc0\x00\x00\xfaD\x00\x00\xfaD\x00\x00\x80?\x00\x00\x00\x00\x9a\x99\x99>\x9a\x99\x99>\n\xd7\xa3<\n\xd7\xa3<l\xe7\x17N\x00|\x12I\x00$\xf4I\x7f\x10\xa7\xc4\x00\x00\x00\x00\t\x9f;=\x00\xb6\x12?\xb6dX=\x00\x00 @\xe1z\x04@33\x03A\xcd\xcc\xcc=\x00\x00\x00\x00\x00\x00\x00\x00\xc4\x00\x80?\x17z\x1e<\x01\x00\x80?;\xc4\x16:\x00\x00\x00\x00\x00\x00\x00\x00\xcd\xcc\x16B\xbf\xbb 9\xc4.\x16Ao\x12\x03;\xe0\\\xdfP\x00\x00\x80?@Q\n?\xc3\xf5\xc8?}\x1d\x90&1\x08\xac<\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xa0A\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x19\x86>\x89[f?\x00\x00\x00\x00\xb1\xfd\x84?\xc4\xf2\x8a>7\t\tC\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xa0A\x00\x00\x00\x00\x00\x00\x00\x00\xcd\xcc\xcc=\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80>\x00\x00\x00?\x00\x00@?\x00\x00\x80?\x00\x00\x00?\xcd\xccL>\xfc(m?\x92\xe1\x8f?\x9a\x99\x15C\x9a\x99\x15CA/q>\xaa\xf1R>q=\n>\xf8Sc>\x00\x00\x00?\x00\x00\x00\x00|a"?\xdaX1?\xce\xaa_?\x0c\x93)?\x8bl\xc7>}]\x1c@\x17\x93i?\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80?\x00P\xc3G\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\t\xca>\xecQ8?\xfc\x18s?\xc3\xf5\x88\xbf\xe1z\x04@\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00o\xf0%?=,\x04?\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00h\x91-?\xe7\xfbi?\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00@\x9cG\x00\x00\x00\x00\x00\x00\x00\x00\x805=I\x80j\xc9I\x80\x96\x98J\xa07 K,\x91\x10\xc3\xb5)C\xc2x\xbd\xb9>>*\xdf\xb9\xd9\xd8?:&\x02\x95D\xf8\xd7\x91D\xcb\x9e\xafD\x01\x02RD\x00\x00\x80?\xa6\xb8rI<\x07GI\xf1N\x1bI:\xa72I\xcbakP\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00HC', 4)
========================= 1 failed, 50 passed in 9.03s =========================
```
Inputs
Python 3.11.11

- python-pytest@8.4.1
- python-scipy@1.12.0
- python-setuptools@80.9.0
- python-setuptools-scm@8.3.1
- python-numpy@1.26.4
- python-zstandard@0.23.0

Full log:

[xll55wqyh23ayy5k77vbyamicyvglp-python-corsikaio-0.6.0.drv.gz](https://github.com/user-attachments/files/24201651/xll55wqyh23ayy5k77vbyamicyvglp-python-corsikaio-0.6.0.drv.gz)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -19,7 +19,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.11", "3.12", "3.13", "3.14"]
+        include:
+          - python-version: "3.11"
+            extra_requirements: "numpy~=1.26"
+          - python-version: "3.12"
+          - python-version: "3.13"
+          - python-version: "3.14"
 
     steps:
       - uses: actions/checkout@v6
@@ -32,10 +37,12 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
+        env:
+          EXTRA_REQUIREMENTS: ${{ matrix.extra_requirements }}
         run: |
           python --version
           pip install codecov pytest-cov flake8
-          pip install -e .[all]
+          pip install -e .[all] ${EXTRA_REQUIREMENTS}
           pip freeze
 
       - name: Static codechecks

--- a/tests/test_io.py
+++ b/tests/test_io.py
@@ -146,7 +146,7 @@ def test_versions():
     from corsikaio.constants import RUNH_VERSION_POSITION
     from corsikaio.constants import EVTH_VERSION_POSITION
 
-    for path, version in zip(test_files, (6.5, 7.41)):
+    for path, version in zip(test_files, (np.float32(6.5), np.float32(7.41))):
         buffer_size = read_buffer_size(path)
 
         with open(path, 'rb') as f:

